tests/utils_test.py

```diff
@@ -53,3 +57,18 @@ def test_parse_completions_function_call(
     assert (
         function_call == expected
     ), f"actual function call: {function_call}, expected function call: {expected}"
+
+
+def test_trim_state():
+    for i in range(4):
+        with open(f"tests/fixtures/large_state_{i}.json", "r") as f:
+            original_state = json.load(f)
+        state = copy.deepcopy(original_state)
+        trimmed_state = trim_state(state, state["context_trimming_threshold"])
+        assert trimmed_state != original_state, f"state {i} is not trimmed"
+        assert len(json.dumps(original_state)) > len(
+            json.dumps(trimmed_state)
+        ), f"state {i} is not trimmed"
+        assert (
```

Comment:
I think this is the only assertion that matters. If it passes and the original state files are `> 1_000_000`, then the other assertions must be true